From a64f017ad1eefc05d1c53d3d87871fe6f7fc6f3e Mon Sep 17 00:00:00 2001
From: Yi-Chia Hsieh <Yi-Chia.Hsieh@mediatek.com>
Date: Fri, 15 Aug 2025 17:13:14 -0700
Subject: [PATCH 121/128] mtk: mac80211: Reject TWT teardown with all=1

TWT teardown with all=1 requires the TWT scheduling AP to teardown all TWT
the sending STA joined. Reject the request because this is not implemented yet.

Signed-off-by: Yi-Chia Hsieh <Yi-Chia.Hsieh@mediatek.com>
---
 net/mac80211/s1g.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/net/mac80211/s1g.c b/net/mac80211/s1g.c
index 70a4d079..dc752147 100644
--- a/net/mac80211/s1g.c
+++ b/net/mac80211/s1g.c
@@ -124,6 +124,9 @@ ieee80211_s1g_rx_twt_teardown(struct ieee80211_sub_if_data *sdata,
 	struct ieee80211_mgmt *mgmt = (struct ieee80211_mgmt *)skb->data;
 	struct ieee80211_twt_teardown *td = (void *)mgmt->u.action.u.s1g.variable;
 
+	if (td->all)
+		return; /* Not supported yet */
+
 	drv_twt_teardown_request(sdata->local, sdata, &sta->sta, td);
 }
 
-- 
2.45.2

